daemon/daemon.go

```diff
@@ -74,13 +74,15 @@ func (d *Daemon) InstallDriver(language string, image string, update bool) error
 			return ErrLanguageDetection.New()
 		}
 	}
-	if !update {
-		s, err := d.getDriverImage(language)
-		switch {
-		case err == nil:
-			return ErrAlreadyInstalled.New(language, s.Name())
-		case !ErrMissingDriver.Is(err):
-			return ErrRuntime.Wrap(err)
+
+	s, err := d.getDriverImage(language)
+	if !ErrMissingDriver.Is(err) && !update {
+		return ErrRuntime.Wrap(err)
+	}
+	if err == nil && update {
+		err = d.runtime.RemoveDriver(s)
```

Comment:
If `s` is a driver image ID (not a name), then delete should happen after the successful install of a new driver.
But this will lead to the same weird behavior if delete fails. So please add a `TODO` that this should happen after the install and we will take care of it later.

Reply:
Thanks! will add this!